p2p/src/net/mock/transport/tcp/adapter/noise.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use async_trait::async_trait;
+use snowstorm::NoiseStream;
+use tokio::net::TcpStream;
+
+use crate::{error::P2pError, net::mock::peer::Role};
+
+use super::StreamAdapter;
+
+static NOISE_HANDSHAKE_PATTERN: &str = "Noise_XX_25519_ChaChaPoly_SHA256";
+
+static NOISE_HANDSHAKE_PARAMS: once_cell::sync::Lazy<snowstorm::NoiseParams> =
+    once_cell::sync::Lazy::new(|| NOISE_HANDSHAKE_PATTERN.parse().expect("valid pattern"));
+
+#[derive(Debug)]
+pub struct NoiseEncryptionAdapter {}
+
+pub struct StreamConfig {
+    local_key: snowstorm::Keypair,
+}
+
+impl Default for StreamConfig {
```

Comment:
There seems to be a misunderstanding regarding the workflow with keys here.

First, there's no default. Well, there's default for configuration typically, but you can't call "keys" configuration, then generate new keys on every instantiation by calling it a default.

Keys can come from two sources:
- From a file, as in from the data directory for someone who already initialized their node
- New key, which you're calling default.

In my opinion, instead of calling this configuration, you should call it "keys". So something like StreamKey, then you don't use "Default" for that at all. Instead, you either generate a new key, or you load it from a file.

Reply:
Sure. For now only `gen_new` is added, but we can add later loading from file as well. Default is removed.